test/unit/data-types/enum.test.ts

```diff
@@ -0,0 +1,25 @@
+import { DataTypes } from '@sequelize/core';
+
+describe('DataTypes.ENUM', () => {
```

Comment:
we already have a unit test suite for DataTypes.ENUM in `misc-data-types.test.ts`, this should be moved there or vice-versa

Reply:
Let's move the ENUM tests from `misc-data-types.test.ts` to this file

Reply:
Tests that were moved from there to this file should also be removed